src/plugins/orangehrmAdminPlugin/Service/LocalizationService.php

```diff
@@ -270,4 +275,165 @@ private function getXliffXmlSources(array $i18nGroups, I18NLanguage $language):
+        $document->loadXML($content);
+
+        if (null !== $targetLanguage = $this->getTargetLanguageFromFile($document)) {
+            $normalizedLocalePattern = sprintf(
+                '(%s|%s)',
+                preg_quote($targetLanguage, '/'),
+                preg_quote(str_replace('-', '_', $targetLanguage), '/')
+            );
+            // strict file names require translation files to be named '____.locale.xlf'
+            $expectedFilenamePattern = sprintf('/^.*\.(?i:%s)\.(?:xlf|xliff)/', $normalizedLocalePattern);
+
+            if (0 === preg_match($expectedFilenamePattern, basename($file))) {
+                $errors[] = [
+                    'line' => -1,
+                    'column' => -1,
+                    'message' => sprintf(
+                        'There is a mismatch between the language included in the file name ("%s") and the "%s" value used in the "target-language" attribute of the file.',
+                        basename($file),
+                        $targetLanguage
+                    ),
+                ];
+            }
+        }
+
+        foreach (XliffUtils::validateSchema($document) as $xmlError) {
+            $errors[] = [
+                'line' => $xmlError['line'],
+                'column' => $xmlError['column'],
+                'message' => $xmlError['message'],
+            ];
+        }
+
+        libxml_clear_errors();
+        libxml_use_internal_errors($internal);
+
+        return ['file' => $file, 'isValid' => 0 === \count($errors), 'messages' => $errors];
+    }
+
+    private function getTargetLanguageFromFile(\DOMDocument $xliffContents): ?string
+    {
+        foreach ($xliffContents->getElementsByTagName('file')[0]->attributes ?? [] as $attribute) {
+            if ('target-language' === $attribute->nodeName) {
+                return $attribute->nodeValue;
+            }
+        }
+
+        return null;
+    }
+
+    public function validateXliffSourceAndTarget($xmlContent): array
+    {
+        // Load the XLIFF content into a DOMDocument
+        $xliffDocument = new \DOMDocument();
+        $xliffDocument->loadXML($xmlContent);
+
+        // Initialize an array to store validation errors
+        $errors = [];
+        $locale = "en_US";
+
+        // Get all the <unit> elements from the XLIFF document
+        $units = $xliffDocument->getElementsByTagName('unit');
+
+        // Define validation constraints, need to be finalized
+        $maxLength = 255000;
+
+        // Validate each <unit> element
+        foreach ($units as $unit) {
+            $unitId = $unit->getAttribute('id');
+            $source = $unit->getElementsByTagName('source')->item(0)->nodeValue;
+            $target = $unit->getElementsByTagName('target')->item(0)->nodeValue;
+
+            // Initialize an array to store errors for this unit
+            $unitErrors = "";
+
+            // Validate source text
+            if (strlen($source) > $maxLength) {
```

Comment:
```suggestion
            if ($this->getTextHelper()->strLength($source, '8bit') > $maxLength) {
```
reference: https://github.com/orangehrm/orangehrm/blob/main/src/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Base64Attachment.php#L98